### PR TITLE
kd-production-review - Fixed & Updated

### PR DESCRIPTION
**Pull request by kasunben**
Better connection logic in the client
Implementing better start functionality for client side store  
Add _authorize method to few other server method calls  
Update Test Cases

### CODE REVIEW

File: lib/client/connect.js [hunk 1]
```diff
@@ -1,19 +1,59 @@
...
-var accessToken = (debug)? (getQueryParamFromURI("access_token") || null) : null;
-Meteor.subscribe('kadira.debug.client.auth', accessToken);
+var env = Utils.getAppEnv();
 
-Meteor.subscribe('kadira.debug.client.listeners');
+var debug = getQueryParamFromURI('kadira_debug');
+var accessToken = (debug)? (getQueryParamFromURI('access_token') || null) : null;
+
+if(env === 'production' && !accessToken) {
+  throw new Meteor.Error('401', 'Unauthorized.');
```
Comment (arunoda) [inline lib/client/connect.js]: No we don't need to throw an error. This is the normal situation. We need to return.

File: lib/client/connect.js [hunk 2]
```diff
@@ -1,19 +1,59 @@
...
+        KadiraDebug._init(browserId, clientId);
+      }
+      
+      Meteor.call('kadira.debug.client.getBrowserName', navigator.userAgent, setName);
+    } else {
+      // invoke the starting process
+      KadiraDebug._init(browserId, clientId);
+    }
+  },
+  onError: function () {
```
Comment (arunoda) [inline lib/client/connect.js]: This should be onStop with `err` as the first arguments.
if `err` is exists that's an error. If not, that's some other disconnection.

File: lib/client/store.js [hunk 3]
```diff
@@ -28,6 +30,12 @@ Store.prototype.start = function(browserId, clientId) {
     throw new Error("browserId or clientId can't be empty.");
   }
 
+  // if Store already started
+  // return without do anything
+  if(this.isStarted) {
```
Comment (arunoda) [inline lib/client/store.js]: Actually, you need to create a method called `this._isStarted()`

**Comment by arunoda** [comment]
We need to authorize [here](https://github.com/kadirahq/meteor-debug/blob/kd-production-review/lib/server/connect.js#L181) as well. Don't we?

**Comment by kasunben** [comment]
Updated.

**Comment by kasunben** [comment]
@arunoda didn't authorize getTrace method because it's also authorizing on trace store when ask the traces. It allow to get trace only for registered sessions.

**Comment by arunoda** [comment]
We need to have the authorize function for `getTrace` as well.
Tests are also failing, check that as well.

**Comment by kasunben** [comment]
Updated with authorized getTrace() method

**Comment by arunoda** [comment]
This is great.
Let's take this in :+1:

**Pull request merged**